### PR TITLE
removes the stupid species check from potion of flight

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

removes the stupid species check from potion of flight

## Why It's Good For The Game

wings work on all species now instead of only having sprites for some things or far far back in the past only having wings as a species feature (angels)

i really don't like how anyone can have wings but now we must add a ton of checks so that only some do. the check itself is getting bloaty as more and more wings are added and its currently detracting from adding more

## Changelog
:cl:
balance: removes the stupid species check from potion of flight
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
You should probably leave the bit in that prevents moths with burnt off wings from getting full wings.

**Comment by JohnFulpWillard** [comment]
I was thinking if it’s was possible to just have a check if your species has wings, because plasmaman with wings doesn’t really make sense

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 1]
```diff
@@ -629,7 +629,7 @@
 	if(iscarbon(exposed_mob) && exposed_mob.stat != DEAD)
 		var/mob/living/carbon/exposed_carbon = exposed_mob
 		var/holycheck = ishumanbasic(exposed_carbon)
-		if(reac_volume < 5 || !(holycheck || islizard(exposed_carbon) || (ismoth(exposed_carbon) && exposed_carbon.dna.features["moth_wings"] != "Burnt Off"))) // implying xenohumans are holy //as with all things,
```
Comment (JohnFulpWillard) [inline code/modules/mining/lavaland/necropolis_chests.dm:632]: Why remove the holy check too?

Comment (Fikou) [reply code/modules/mining/lavaland/necropolis_chests.dm:632]: if he didnt remove holy check only regular humans would be allowed

Comment (tralezab) [reply code/modules/mining/lavaland/necropolis_chests.dm:632]: it's also stupid

**Review by coiax — COMMENTED** [review]
Kind of just want to beg for you to clean up this `expose_mob` code in general, it's REALLY hard to follow.

But I know that's out of scope 🤷

**Comment by tralezab** [comment]
I'll take a look but no promises, if I'm up to it the codebase will gain more from me fixing that code up than it will lose in an out of scope pr

**Comment by Rohesie** [comment]
Which species would gain the ability to get wings with this?

A better code solution for this would be to have a whitelist in the form of a flag or trait, instead of multiple istype() blacklist checks.

**Comment by Sheits** [comment]
just wanna say we should probably merge something that fixes this check sooner rather than later coz the pr mentioned wont work without this, but it shouldnt cause any bugs ingame, just wont be able to see any of the newer wings until then.

**Comment by tralezab** [comment]
> Which species would gain the ability to get wings with this?
> 
> A better code solution for this would be to have a whitelist in the form of a flag or trait, instead of multiple istype() blacklist checks.

all of them, any species can have wings, it's just that not all have unique wing sprites. i think this is no longer a good reason to block flight.

**Comment by Rohesie** [comment]
> 
> 
> > Which species would gain the ability to get wings with this?
> > A better code solution for this would be to have a whitelist in the form of a flag or trait, instead of multiple istype() blacklist checks.
> 
> all of them, any species can have wings, it's just that not all have unique wing sprites. i think this is no longer a good reason to block flight.

Let me rephrase the question: which species that didn't have flight will now gain it?

**Comment by Time-Green** [comment]
> > > Which species would gain the ability to get wings with this?
> > > A better code solution for this would be to have a whitelist in the form of a flag or trait, instead of multiple istype() blacklist checks.
> > 
> > 
> > all of them, any species can have wings, it's just that not all have unique wing sprites. i think this is no longer a good reason to block flight.
> 
> Let me rephrase the question: which specific species that didn't have flight will now gain it? Do you know the extent of your changes, exactly?

Every species. It used to be unique to only humans, moths and lizards. It was only locked to those three species because the other races didn’t have sprites for the wings (most still don’t).

**Comment by Rohesie** [comment]
> Every species. It used to be unique to only humans, moths and lizards. It was only locked to those three species because the other races didn’t have sprites for the wings (most still don’t).
> 
> Right now those spriteless species will default to angel wings.

Yes, I understand it's every species that didn't have wings. Which species are those exactly? Can you name them?

**Comment by Time-Green** [comment]
nightmare
shadowperson
abductor
android
dullahan
ethereal
felinid
fly
iron golem
adamantine golem
plasma golem
diamond golem
gold golem
silver golem
plasteel golem
titanium golem
plastitanium golem
alloy golem
wood golem
uranium golem
sand golem
glass golem
bluespace golem
bananium golem
runic golem
cloth golem
plastic golem
bronze golem
cardboard golem
leather golem
durathread golem
bone golem
snow golem
Metallic Hydrogen golem
slime person
luminescent
stargazer slime person
jellyperson
monkey
mushroom persoon
plasmaman
podperson
skeleton
snail
military synthethic
synthethic
vampire
meme zombie
zombie
meth zombie

**Comment by Rohesie** [comment]
Snails with wings? Vampires with wings? Monkeys? Flies with angel wings? Golems as well?
I see several I'd be in doubt of, and I think it'd be better as a whitelist instead.
Let's see what other maintainers think.

**Comment by Time-Green** [comment]
> Snails with wings? Vampires with wings? Monkeys? Flies with angel wings? Golems as well?
> I see several I'd be in doubt of, and I think it'd be better as a whitelist instead.
> Let's see what other maintainers think.

Flies got their own sprite, they’re just disabled until this PR is merged

**Comment by tralezab** [comment]
IMO: it's better to give races angel wings than to block a huge gameplay buff based on whether a species has sprites for it

**Comment by Rohesie** [comment]
> 
> 
> IMO: it's better to give races angel wings than to block a huge gameplay buff based on whether a species has sprites for it

I don't really see why every species should have this huge gameplay buff.
Get the support of a maintainer in order to make this species change, please, before re-opening it.

**Comment by EricZilla** [comment]
> > IMO: it's better to give races angel wings than to block a huge gameplay buff based on whether a species has sprites for it
> 
> I don't really see why every species should have this huge gameplay buff.
> Get the support of a maintainer in order to make this species change, please, before re-opening it.

Because you drink a magic potion to give you wings, humans arent birds but they get feathers, having it be angel wings by default until someone decides to add a special one seems just fine to me. Also if we wanna talk about realism of a species having wings, consider that humans should have lemur wings
